src/app.js

```diff
@@ -1 +1,124 @@
-console.log('Hallo daar!');
\ No newline at end of file
+import axios from "axios";
+
+const countryList = document.getElementById('countries');
+
+function colorCountries(region) {
+    let continentColor = '';
+
+    switch (region) {
+        case 'Asia':
+            continentColor = 'continent-list-asia';
+            break;
+        case 'Americas':
+            continentColor = 'continent-list-americas';
+            break;
+        case 'Africa':
+            continentColor = 'continent-list-africa';
+            break;
+        case 'Europe':
+            continentColor = 'continent-list-europe';
+            break;
+        case 'Oceania':
+            continentColor = 'continent-list-oceania';
+            break;
+        default:
+            continentColor = 'continent-list-default';
+    }
+    return continentColor;
+};
+
+async function fetchCountries() {
+    try {
+        // het request maken
+        const response = await axios.get('https://restcountries.com/v2/all');
+        console.log(response.data);
+
+        response.data.sort((a, b) => a.population - b.population);
+
+        const listItems = response.data.map(({name, population, region, flag}) => {
+            let color = colorCountries(region);
+            return `<li class="${color}"><img src="${flag}" alt="Flag" width="25"> ${name}</li>
+            <li class="countries-list">Has a population of ${population} people</li>`
+        });
+        countryList.innerHTML = listItems.join('');
+
+    } catch (e) {
+        // de errors afvangen
+        console.error(e);
+    }
+};
+
+fetchCountries();
+
+
+//testen voor 1 entry
+// countryList.innerHTML = `
+// <li> <span> ${response.data[0]} </span> <span> Has a population of ${response.data[0]} people</span></li>
+// `;
+
+
+// PART 2
+
+const countryData = document.getElementById('country-searcher-output');
+
+const searchForm = document.getElementById('search-form');
+searchForm.addEventListener('submit', searchName);
+
+function searchName(e) {
+    e.preventDefault();
+
+    const searchInput = document.getElementById('search-input');
+
+    searchCountries(searchInput.value);
+
+    searchInput.value = '';
+
+}
+
+
+async function searchCountries(name) {
+    try {
+        const output = await axios.get(`https://restcountries.com/v2/name/${name}`);
+        const country = output.data[0];
```

Comment:
Je zou hier ook het volgende van kunnen maken
```suggestion
        const { flag, name, population, region, capital, currencies, languages } = output.data[0];
```
Dan hoef je vervolgens nergers meer `country.` vóór te zetten